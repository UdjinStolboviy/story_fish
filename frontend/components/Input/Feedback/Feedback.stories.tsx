import { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { expect } from "@storybook/jest";
import { screen } from "@storybook/testing-library";

import { Feedback } from "./Feedback";

export default {
  title: "Controls/Feedback",
  component: Feedback,
} as ComponentMeta<typeof Feedback>;

export const ValidFeedback: ComponentStoryObj<typeof Feedback> = {
  play: async () => {
    await expect(screen.getByText("Looks good!")).toBeInTheDocument();
  },
};
ValidFeedback.args = {
  children: "Looks good!",
  isValid: true,
};

export const InvalidFeedback: ComponentStoryObj<typeof Feedback> = {
  play: async () => {
    await expect(
      screen.getByText("Please provide a valid value")
    ).toBeInTheDocument();
  },
};
InvalidFeedback.args = {
  children: "Please provide a valid value",
  isValid: false,
};
