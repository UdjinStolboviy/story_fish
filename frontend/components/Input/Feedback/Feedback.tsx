import { FC } from "react";

import styled from "@emotion/styled";

type Props = {
  isValid?: boolean;
};

export const Feedback = styled.span<Props>`
  color: ${({ isValid, theme }) =>
    isValid ? theme.font.valid : theme.font.invalid};
`;

export const ConditionalFeedback: FC = ({ children }) =>
  children ? <Feedback>{children}</Feedback> : <>&nbsp;</>;
