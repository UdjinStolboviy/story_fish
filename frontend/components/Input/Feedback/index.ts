export { Feedback, ConditionalFeedback } from "./Feedback";
