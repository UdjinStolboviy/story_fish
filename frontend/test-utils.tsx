import { FC, ReactElement } from "react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { render, RenderOptions } from "@testing-library/react";
import { ThemeProvider } from "@emotion/react";

import { rootReducer, RootState } from "./store";
import { Themes } from "./styles/themes";
import { Layout } from "./components/Layout";

type Options = { preloadedState?: RootState } & RenderOptions;

const customRender = (
  ui: ReactElement,
  { preloadedState, ...options }: Options = {}
) => {
  const store = configureStore({ reducer: rootReducer, preloadedState });

  const Wrapper: FC = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider theme={Themes.light}>{children}</ThemeProvider>
    </Provider>
  );
  return render(ui, { wrapper: Wrapper, ...options });
};

const pageRender = (
  ui: ReactElement,
  { preloadedState, ...options }: Options = {}
) => {
  const store = configureStore({ reducer: rootReducer, preloadedState });

  const Wrapper: FC = ({ children }) => (
    <Provider store={store}>
      <Layout>{children}</Layout>
    </Provider>
  );
  return render(ui, { wrapper: Wrapper, ...options });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, pageRender };
