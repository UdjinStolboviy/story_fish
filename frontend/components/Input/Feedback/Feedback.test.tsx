import { render } from "@/test-utils";

import { Feedback } from "./Feedback";

describe("Feedback test cases", () => {
  it("Render check for valid", () => {
    const { asFragment } = render(<Feedback isValid>Looks good!</Feedback>);

    expect(asFragment()).toMatchSnapshot();
  });
  it("Render check for invalid", () => {
    const { asFragment } = render(
      <Feedback>Please provide a valid value</Feedback>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
