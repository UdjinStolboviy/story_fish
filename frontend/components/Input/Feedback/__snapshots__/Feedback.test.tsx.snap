// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback test cases Render check for invalid 1`] = `
<DocumentFragment>
  <span
    class="css-sk8pyf"
  >
    Please provide a valid value
  </span>
</DocumentFragment>
`;

exports[`Feedback test cases Render check for valid 1`] = `
<DocumentFragment>
  <span
    class="css-3xrj9w"
  >
    Looks good!
  </span>
</DocumentFragment>
`;
